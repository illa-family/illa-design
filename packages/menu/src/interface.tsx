import {
  ForwardRefExoticComponent,
  HTMLAttributes,
  MouseEvent,
  PropsWithChildren,
  ReactNode,
} from "react"
import { TriggerProps } from "@illa-design/trigger"
import { SerializedStyles } from "@emotion/serialize"

export type MenuVariant = "inline" | "pop"
export type Theme = "light" | "dark"
type Mode = "vertical" | "horizontal" | "popButton"

export interface MenuProps extends HTMLAttributes<HTMLDivElement> {
  theme?: Theme
  mode?: Mode
  variant?: MenuVariant
  levelIndent?: number
  accordion?: boolean
  collapseDefaultIcon?: ReactNode
  collapseActiveIcon?: ReactNode
  autoOpen?: boolean
  hasCollapseButton?: boolean
  collapse?: boolean
  selectable?: boolean
  ellipsis?: boolean
  inDropdown?: boolean
  defaultSelectedKeys?: string[]
  defaultOpenKeys?: string[]
  selectedKeys?: string[]
  openKeys?: string[]
  isMenu?: boolean
  triggerProps?: Partial<TriggerProps>
  onClickMenuItem?: (key: string, event: MouseEvent, keyPath: string[]) => any
  onClickSubMenu?: (key: string, openKeys: string[], keyPath: string[]) => void
  onCollapseChange?: (collapse: boolean) => void
}

export interface IndentProps {
  level?: number
  levelIndent?: number
}

export interface ItemProps extends Omit<HTMLAttributes<HTMLElement>, "title"> {
  key: string
  _key?: string
  _css?: SerializedStyles
  title?: string | ReactNode
  disabled?: boolean
  level?: number
  needTooltip?: boolean
}

export interface ItemGroupProps {
  title?: string | ReactNode
  level?: number
  _css?: SerializedStyles
  children?: ReactNode
}

export interface SubMenuProps {
  key: string
  _key?: string
  _css?: SerializedStyles
  title?: string | ReactNode
  level?: number
  selectable?: boolean
  children?: ReactNode
}

export interface OverflowWrapperProps {
  children?: ReactNode
}

export interface MenuComponent
  extends ForwardRefExoticComponent<PropsWithChildren<MenuProps>> {
  Item: ForwardRefExoticComponent<ItemProps>
  ItemGroup: ForwardRefExoticComponent<PropsWithChildren<ItemGroupProps>>
  SubMenu: ForwardRefExoticComponent<PropsWithChildren<SubMenuProps>>
}
