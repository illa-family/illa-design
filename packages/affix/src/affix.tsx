import {
  forwardRef,
  useCallback,
  useEffect,
  useImperativeHandle,
  useRef,
  useState,
} from "react"
import { isFunction, throttleByRaf } from "@illa-design/system"
import useMeasure from "react-use/lib/useMeasure"
import useIsomorphicLayoutEffect from "react-use/lib/useIsomorphicLayoutEffect"
import { applyAffixFixedStyle, applySize } from "./style"
import { AffixProps } from "./interface"

enum AffixStatus {
  START,
  DONE,
}

function getTargetRect(target: HTMLElement | Window): DOMRect {
  return target !== window
    ? (target as HTMLElement).getBoundingClientRect()
    : ({ top: 0, bottom: window.innerHeight } as DOMRect)
}

export const Affix = forwardRef<HTMLDivElement, AffixProps>((props, ref) => {
  const {
    offsetTop = 0,
    offsetBottom,
    target = () => window,
    targetContainer,
    onChange,
    children,
    ...rest
  } = props

  const [status, setStatus] = useState(AffixStatus.DONE)
  const [position, setPosition] = useState<{
    type: "top" | "bottom"
    offset: number
  }>({
    type: "top",
    offset: 0,
  })
  const [size, setSize] = useState<{ width: number; height: number }>({
    width: 0,
    height: 0,
  })

  const mounted = useRef(true)
  const targetRef = useRef<HTMLElement | Window | null>(null)
  const wrapperRef = useRef<HTMLDivElement | null>(null)
  const lastIsFixed = useRef(false)

  const [measureWrapperRef, measureWrapperInfo] = useMeasure<HTMLDivElement>()
  const [measureAffixRef, measureAffixInfo] = useMeasure<HTMLDivElement>()

  const setWrapperRefs = (el: HTMLDivElement) => {
    // for size measure
    measureWrapperRef(el)
    // for position compute
    wrapperRef.current = el
  }

  const updatePosition = useCallback(
    throttleByRaf(() => {
      /*
       * check is mounted to avoid:
       * Warning: Can't perform a React state update on an unmounted component.
       */
      mounted.current && setStatus(AffixStatus.START)
    }),
    [],
  )

  useImperativeHandle(ref, () => wrapperRef?.current as HTMLDivElement, [])

  useEffect(() => {
    mounted.current = true
    return () => {
      mounted.current = false
    }
  }, [])

  useEffect(() => {
    updatePosition()
  }, [
    offsetTop,
    offsetBottom,
    target,
    targetContainer,
    updatePosition,
    measureWrapperInfo,
    measureAffixInfo,
  ])

  useEffect(() => {
    const events = ["scroll", "resize"]

    targetRef.current = target && isFunction(target) ? target() : null

    if (targetRef.current) {
      events.forEach((event) => {
        targetRef.current?.addEventListener(event, updatePosition)
      })

      return () => {
        events.forEach((event) => {
          targetRef.current?.removeEventListener(event, updatePosition)
        })
      }
    }
  }, [target, updatePosition])

  useEffect(() => {
    const container =
      targetContainer && isFunction(targetContainer) ? targetContainer() : null

    if (targetRef.current !== window && container) {
      container.addEventListener("scroll", updatePosition)

      return () => {
        container.removeEventListener("scroll", updatePosition)
      }
    }
  }, [targetContainer, updatePosition])

  useIsomorphicLayoutEffect(() => {
    if (
      status !== AffixStatus.START ||
      !targetRef.current ||
      !wrapperRef.current
    ) {
      return
    }

    const type = offsetBottom === undefined ? "top" : "bottom"
    const wrapperRect = getTargetRect(wrapperRef.current)
    const targetRect = getTargetRect(targetRef.current)

    let newIsFixed = false
    let offset

    if (type === "top") {
      newIsFixed = wrapperRect.top - targetRect.top < (offsetTop || 0)
      offset = targetRect.top + (offsetTop || 0)
    } else {
      newIsFixed = targetRect.bottom - wrapperRect.bottom < (offsetBottom || 0)
      offset = window.innerHeight - targetRect.bottom + (offsetBottom || 0)
    }

    if (newIsFixed !== lastIsFixed.current) {
      lastIsFixed.current = newIsFixed
      onChange && isFunction(onChange) && onChange(newIsFixed)
    }

    setStatus(AffixStatus.DONE)
    setPosition({
      type,
      offset,
    })
    setSize({
      width: wrapperRef.current.clientWidth,
      height: wrapperRef.current.clientHeight,
    })
  })

  return (
    <div ref={setWrapperRefs} {...rest}>
      {lastIsFixed.current && <div css={applySize(size)}></div>}
      <div
        css={applyAffixFixedStyle({
          isFixed: lastIsFixed.current,
          position,
          size,
        })}
        ref={measureAffixRef}
      >
        {children}
      </div>
    </div>
  )
})

Affix.displayName = "Affix"
