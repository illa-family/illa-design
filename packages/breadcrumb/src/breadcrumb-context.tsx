import { createContext } from "react"
import { BreadcrumbContextProps } from "./interface"

export const BreadcrumbContext = createContext<BreadcrumbContextProps>({
  isCurrent: false,
  path: "",
  breadcrumbName: "",
  children: [],
})

BreadcrumbContext.displayName = "BreadcrumbContext"
