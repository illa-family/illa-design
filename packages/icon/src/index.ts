export * from "./icon"
export * from "./interface"
export * from "./create-icon"
// icons
export * from "./icons/close"
export * from "./icons/person"
export * from "./icons/image-default"
export * from "./icons/loading"
export * from "./icons/copy"
export * from "./icons/checkmark"
export * from "./icons/eye-on"
export * from "./icons/eye-off"
export * from "./icons/error"
export * from "./icons/search"
export * from "./icons/link"
export * from "./icons/right"
export * from "./icons/error"
export * from "./icons/warning"
export * from "./icons/success"
export * from "./icons/reduce"
export * from "./icons/empty"
export * from "./icons/warning-circle"
export * from "./icons/info-cricle"
export * from "./icons/pre"
export * from "./icons/next"
export * from "./icons/up"
export * from "./icons/down"
export * from "./icons/more"
export * from "./icons/add"
export * from "./icons/expand"
export * from "./icons/upload"
export * from "./icons/minus"
export * from "./icons/plus"
export * from "./icons/delete"
export * from "./icons/file-word"
export * from "./icons/file-pdf"
export * from "./icons/file-default"
export * from "./icons/file-excel"
export * from "./icons/file-music"
export * from "./icons/file-ppt"
export * from "./icons/file-picture"
export * from "./icons/file-video"
export * from "./icons/file-wps"
export * from "./icons/like"
export * from "./icons/share"
export * from "./icons/star"
export * from "./icons/heart"
export * from "./icons/caret-down"
export * from "./icons/drag-point"
export * from "./icons/pen"
export * from "./icons/pre-double"
export * from "./icons/next-double"
export * from "./icons/leaf"
export * from "./icons/caret-left"
export * from "./icons/caret-right"
export * from "./icons/sorter-default"
export * from "./icons/sorter-up"
export * from "./icons/sorter-down"
export * from "./icons/filter"
export * from "./icons/time"
export * from "./icons/calendar"
export * from "./icons/caret-down"
export * from "./icons/caret-left"
export * from "./icons/caret-right"
export * from "./icons/info"
