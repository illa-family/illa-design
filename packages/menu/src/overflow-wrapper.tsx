import React, {ReactElement, ReactNode, useEffect, useRef, useState,} from "react"
import {getStyle} from "@illa-design/system"
import useMeasure from "react-use/lib/useMeasure"
import {OverflowWrapperProps} from "./interface"
import {SubMenu} from "./sub-menu"
import {overflowMenuItemCss} from "./style"
import {overflowWrapperCss, subMenuPlaceholderCss} from "./styles"

function getNodeWidth(node: HTMLElement) {
  if (!node) {
    return 0
  }

  const width = +node.getBoundingClientRect().width.toFixed(2)
  const marginLeft = px2Number(getStyle(node, "margin-left"))
  const marginRight = px2Number(getStyle(node, "margin-right"))

  return width + marginLeft + marginRight
}

function px2Number(str: string): number {
  const num = Number(str.replace("px", ""))
  return Number.isNaN(num) ? 0 : num
}

export const OverflowWrapper = (props: OverflowWrapperProps) => {
  const { children } = props
  const wrapperRef = useRef<HTMLDivElement | null>(null)
  const [measureWrapperRef, measureWrapperInfo] = useMeasure<HTMLDivElement>()
  const [lastVisibleIndex, setLastVisibleIndex] = useState<number | null>(null)
  const OVERFLOW_THRESHOLD = 10

  useEffect(() => {
    calcLastVisibleIndex()
  }, [children, measureWrapperInfo])

  function calcLastVisibleIndex() {
    if (!wrapperRef.current) {
      return
    }

    const wrapperElement = wrapperRef.current
    const maxWidth = getNodeWidth(wrapperElement) - OVERFLOW_THRESHOLD
    const childrenNodeList = Array.from(wrapperElement.children)

    let menuItemIndex = 0
    let currentMaxWidth = 0
    let overflowSubMenuWidth = 0

    for (const node of childrenNodeList) {
      if (node.getAttribute("data-sub-menu-marker") !== null) {
        if (node.getAttribute("data-sub-menu-placeholder-marker")) {
          overflowSubMenuWidth = getNodeWidth(node as HTMLElement)
        }
        continue
      }
      const nodeWidth = getNodeWidth(node as HTMLElement)
      currentMaxWidth += nodeWidth

      if (currentMaxWidth > maxWidth) {
        menuItemIndex--
        currentMaxWidth -= nodeWidth

        if (currentMaxWidth + overflowSubMenuWidth > maxWidth) {
          menuItemIndex--
        }

        setLastVisibleIndex(menuItemIndex)
        return
      }

      menuItemIndex++
    }

    setLastVisibleIndex(null)
  }

  const renderOverflowSubMenu = (
    children: ReactNode,
    isPlaceholder: boolean,
  ) => {
    return (
      <SubMenu
        title={<span>...</span>}
        key={`overflow-sub-menu${isPlaceholder && "placeholder"}`}
        data-sub-menu-marker
        data-sub-menu-placeholder-marker={isPlaceholder}
        css={isPlaceholder && subMenuPlaceholderCss}
        {...props}
        children={children}
      />
    )
  }

  const renderChildren = () => {
    let overflowSubMenu = null
    const overflowSubMenuPlaceholder = renderOverflowSubMenu(null, true)

    const originalMenuItems = React.Children.map(children, (child, index) => {
      let item = child

      if (lastVisibleIndex !== null) {
        // set overflow item invisible
        if (index > lastVisibleIndex) {
          item = React.cloneElement(child as ReactElement, {
            _css: overflowMenuItemCss,
          })
        }

        // put overflow item into sub menu
        if (index === lastVisibleIndex + 1) {
          const overflowMenuItems = React.Children.toArray(children)
            .slice(lastVisibleIndex + 1)
            .map((child) => React.cloneElement(child as ReactElement), {
              key: (child as ReactElement).props._key,
            })

          overflowSubMenu = renderOverflowSubMenu(overflowMenuItems, false)
        }
      }

      return item
    })

    return [
      overflowSubMenuPlaceholder,
      ...(originalMenuItems as ReactElement[]),
      overflowSubMenu,
    ]
  }

  return (
    <div
      ref={(el) => {
        wrapperRef.current = el
        measureWrapperRef(el as HTMLDivElement)
      }}
      css={overflowWrapperCss}
    >
      {renderChildren()}
    </div>
  )
}
