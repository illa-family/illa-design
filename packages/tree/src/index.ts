export * from "./tree"
export * from "./interface"
export * from "@illa-design/tree-common"
