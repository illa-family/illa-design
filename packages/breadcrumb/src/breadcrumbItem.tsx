import { forwardRef, useContext } from "react"
import { BreadcrumbItemProps } from "./interface"
import { itemCss, itemCurrentCss, itemHoverCss } from "./style"
import { css } from "@emotion/react"
import { BreadcrumbContext } from "./breadcrumb-context"
import { Dropdown } from "@illa-design/dropdown"
import { Menu } from "@illa-design/menu"

export const BreadcrumbItem = forwardRef<HTMLDivElement, BreadcrumbItemProps>(
  (props, ref) => {
    const { _css, dropList, dropdownProps, ...restProps } = props

    const { isCurrent, path, breadcrumbName, children } =
      useContext(BreadcrumbContext)

    const { Item } = Menu

    const dropItem = children?.length && (
      <Menu>
        {children.map((cItem: any, cIdx: number) => {
          return <Item key={cIdx + ""} title={cItem.breadcrumbName}></Item>
        })}
      </Menu>
    )

    const finalDrop = dropList || dropItem

    const handleItemCss = () => {
      return css`
        ${itemCss};
        ${isCurrent && itemCurrentCss};
        ${path && itemHoverCss}
      `
    }

    const item = (
      <div ref={ref} {...restProps} css={css(handleItemCss(), _css)}>
        {breadcrumbName || props.children}
      </div>
    )

    return finalDrop ? (
      <Dropdown dropList={finalDrop} {...dropdownProps}>
        {item}
      </Dropdown>
    ) : (
      item
    )
  },
)

BreadcrumbItem.displayName = "BreadcrumbItem"
