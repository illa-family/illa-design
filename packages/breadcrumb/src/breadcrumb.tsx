import { Children, forwardRef } from "react"
import { BreadcrumbProps } from "./interface"
import { css } from "@emotion/react"
import { separatorCss, itemCss, wrapCss } from "./style"
import { BreadcrumbContext } from "./breadcrumb-context"
import { BreadcrumbItem } from "./breadcrumbItem"

export const Breadcrumb = forwardRef<HTMLDivElement, BreadcrumbProps>(
  (props, ref) => {
    const { _css, separator, routes, maxCount, ...restProps } = props

    function Separator() {
      return <span css={separatorCss}>{separator ? separator : "/"}</span>
    }
    const renderItem = (list: any, idx: number) => {
      if (!maxCount || list.length < maxCount) return true
      if (idx === 0 || idx > list.length - maxCount) return true
      return false
    }

    const routeItems =
      routes?.length &&
      routes.map((item, idx: number) => {
        return (
          <>
            <BreadcrumbContext.Provider
              value={{
                isCurrent: idx === routes.length - 1,
                path: item.path,
                breadcrumbName: item.breadcrumbName,
                children: item.children,
              }}
            >
              {renderItem(routes, idx) ? (
                <BreadcrumbItem key={idx} />
              ) : (
                idx === 1 && <span css={itemCss}>...</span>
              )}
            </BreadcrumbContext.Provider>
            {idx < routes.length - 1 &&
              (renderItem(routes, idx) ? (
                <Separator />
              ) : idx === 1 ? (
                <Separator />
              ) : null)}
          </>
        )
      })

    let childLiItem = Children.toArray(props.children)
    const normalItems = Children.map(childLiItem, (ele, idx) => {
      return (
        <>
          <BreadcrumbContext.Provider
            value={{
              isCurrent: idx === childLiItem.length - 1,
            }}
          >
            {renderItem(childLiItem, idx) ? (
              <>{ele}</>
            ) : (
              idx === 1 && <span css={itemCss}>...</span>
            )}
          </BreadcrumbContext.Provider>
          {idx < childLiItem.length - 1 &&
            (renderItem(childLiItem, idx) ? (
              <Separator />
            ) : idx === 1 ? (
              <Separator />
            ) : null)}
        </>
      )
    })

    return (
      <div ref={ref} {...restProps} css={css(wrapCss, _css)}>
        {routes?.length ? routeItems : normalItems}
      </div>
    )
  },
)

Breadcrumb.displayName = "Breadcrumb"
