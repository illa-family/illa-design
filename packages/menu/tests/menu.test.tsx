import { Menu } from "../src"
import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"

const { Item, SubMenu, ItemGroup } = Menu

test("Menu renders with item", () => {
  render(
    <Menu>
      <Item title={"Blog"} key={"1"} disabled />
      <Item title={"Tutorial"} key={"2"} />
      <Item title={"Docs"} key={"3"} />
      <Item title={"Community"} key={"4"} />
      <Item title={"Github"} key={"5"} />
    </Menu>,
  )

  expect(screen.getByText("Blog")).toBeInTheDocument()
  expect(screen.getByText("Docs")).toBeInTheDocument()
})

test("Menu renders with horizontal selected", () => {
  render(
    <Menu defaultSelectedKeys={["1"]} data-testid={"menu"}>
      <Item title={"Blog"} key={"1"} disabled />
      <Item title={"Tutorial"} key={"2"} />
      <Item title={"Docs"} key={"3"} />
      <Item title={"Community"} key={"4"} />
      <Item title={"Github"} key={"5"} />
    </Menu>,
  )

  expect(screen.getByTestId("menu")).toMatchSnapshot()
})

test("Menu renders with item group", () => {
  render(
    <Menu>
      <ItemGroup title={"Group"}>
        <Item title={"Tutorial"} key={"2"} />
        <Item title={"Docs"} key={"3"} />
        <Item title={"Community"} key={"4"} />
        <Item title={"Github"} key={"5"} />
      </ItemGroup>
    </Menu>,
  )

  expect(screen.getByText("Group")).toBeInTheDocument()
  expect(screen.getByText("Docs")).toBeInTheDocument()
})

test("Menu renders with submenu", () => {
  render(
    <Menu hasCollapseButton>
      <ItemGroup title={"Group"}>
        <Item title={"Tutorial"} key={"2"} />
        <Item title={"Docs"} key={"3"} />
      </ItemGroup>
      <SubMenu title={"SubMenu"} key={"0"}>
        <Item title={"Community"} key={"4"} />
        <Item title={"Github"} key={"5"} />
      </SubMenu>
    </Menu>,
  )

  expect(screen.getByText("SubMenu")).toBeInTheDocument()
})

test("Menu renders with overflow wrap", () => {
  const { container } = render(
    <Menu hasCollapseButton mode={"horizontal"}>
      <Item title={"Tutorial"} key={"2"} />
      <Item title={"Docs"} key={"3"} />
      <Item title={"Community"} key={"4"} />
      <Item title={"GitHub"} key={"5"} />
      <Item title={"GitLab"} key={"6"} />
    </Menu>,
  )

  expect(screen.getByText("...")).toBeInTheDocument()
})

test("onClickMenuItem should be trigger", () => {
  const clickEvent = jest.fn()

  render(
    <Menu hasCollapseButton onClickMenuItem={clickEvent}>
      <ItemGroup title={"Group"}>
        <Item title={"Tutorial"} key={"2"} />
        <Item title={"Docs"} key={"3"} />
      </ItemGroup>
      <SubMenu title={"SubMenu"} key={"0"}>
        <Item title={"Community"} key={"4"} />
        <Item title={"Github"} key={"5"} />
      </SubMenu>
    </Menu>,
  )

  screen.getByText("Docs").click()
  expect(clickEvent).toBeCalled()
})

test("onClickMenuItem should NOT be trigger when item is disabled", () => {
  const clickEvent = jest.fn()

  render(
    <Menu hasCollapseButton onClickMenuItem={clickEvent}>
      <ItemGroup title={"Group"}>
        <Item title={"Tutorial"} key={"2"} />
        <Item title={"Docs"} key={"3"} disabled />
      </ItemGroup>
      <SubMenu title={"SubMenu"} key={"0"}>
        <Item title={"Community"} key={"4"} />
        <Item title={"Github"} key={"5"} />
      </SubMenu>
    </Menu>,
  )

  screen.getByText("Docs").click()
  expect(clickEvent).not.toBeCalled()
})

test("onClickSubMenu should be trigger", () => {
  const clickEvent = jest.fn()

  render(
    <Menu hasCollapseButton onClickSubMenu={clickEvent}>
      <ItemGroup title={"Group"}>
        <Item title={"Tutorial"} key={"2"} />
        <Item title={"Docs"} key={"3"} />
      </ItemGroup>
      <SubMenu title={"SubMenu"} key={"0"}>
        <Item title={"Community"} key={"4"} />
        <Item title={"Github"} key={"5"} />
      </SubMenu>
    </Menu>,
  )

  screen.getByText("SubMenu").click()
  expect(clickEvent).toBeCalled()
})

test("onCollapse should be trigger", () => {
  const event = jest.fn()

  const { container } = render(
    <Menu hasCollapseButton onCollapseChange={event}>
      <ItemGroup title={"Group"}>
        <Item title={"Tutorial"} key={"2"} />
        <Item title={"Docs"} key={"3"} />
      </ItemGroup>
      <SubMenu title={"SubMenu"} key={"0"}>
        <Item title={"Community"} key={"4"} />
        <Item title={"Github"} key={"5"} />
      </SubMenu>
    </Menu>,
  )

  ;(container.querySelector(`[data-menu-collapse-icon]`) as HTMLElement).click()

  expect(event).toBeCalled()
})
