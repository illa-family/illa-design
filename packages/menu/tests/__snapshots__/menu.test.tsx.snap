// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders with horizontal selected 1`] = `
.emotion-0 {
  position: relative;
  -webkit-transition: width 0.2s ease-in-out;
  transition: width 0.2s ease-in-out;
  background-color: #fff;
}

.emotion-1 {
  overflow: auto;
  width: 100%;
  height: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 24px;
  font-size: 14px;
  color: #5c5c5c;
  line-height: 40px;
  cursor: not-allowed;
  color: #c2c2c2;
  color: #134ae0;
  background-color: #f5f5f5;
}

.emotion-3 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 24px;
  font-size: 14px;
  color: #5c5c5c;
  line-height: 40px;
}

.emotion-4:hover {
  cursor: pointer;
  background-color: #f5f5f5;
}

<div
  class="emotion-0"
  data-testid="menu"
>
  <div
    class="emotion-1"
  >
    <div
      class="emotion-2"
    >
      <div
        class="emotion-3"
      >
        Blog
      </div>
    </div>
    <div
      class="emotion-4"
    >
      <div
        class="emotion-3"
      >
        Tutorial
      </div>
    </div>
    <div
      class="emotion-4"
    >
      <div
        class="emotion-3"
      >
        Docs
      </div>
    </div>
    <div
      class="emotion-4"
    >
      <div
        class="emotion-3"
      >
        Community
      </div>
    </div>
    <div
      class="emotion-4"
    >
      <div
        class="emotion-3"
      >
        Github
      </div>
    </div>
  </div>
</div>
`;
